horizons/gui/widgets/tooltip.py

```diff
@@ -113,11 +114,38 @@ def get_top(w):
 			self.tooltip_shown = True
 
 		screen_width = horizons.globals.fife.engine_settings.getScreenWidth()
-		self.gui.y = widget_position[1] + self.height + 10
-		xpos = int(widget_position[0] + self.width/2 - self.gui.size[0]/2)
-		xpos = max(xpos, 5)
-		xpos = min(xpos, screen_width - self.gui.size[0] - 5)
+
+		if isinstance(self, ImageButton):
+			# If the a button spawn a tooltip and the cursor hovers over that
+			# tooltip the tooltip will disappear and the button becomes
+			# unclickable. (see issue #2776: https://git.io/vxRrn)
+			# This is a workaround for that problem, by making tooltips for
+			# buttons always display below the button.
+			# There is a small chance that it still happens if you move the
+			# cursor very fast, but that seems unlikely to be such a major
+			# problem as it used to be.
+			# The underlying problem is that the cursor can not be repositioned
+			# on every mousemove because that causes frequent segfaults.
+			# If that problem would be fixed this workaround wouldn't be
+			# neccessary anymore.
+
+			ypos = widget_position[1] + self.height + 10
+
+			xpos = int(widget_position[0] + self.width/2 - self.gui.size[0]/2)
+			xpos = max(xpos, 5)
+			xpos = min(xpos, screen_width - self.gui.size[0] - 5)
```

Comment:
The `min` call prevents the tooltip going off screen on the right.
The `max` call prevents this on the left.
This wasn't necessary before because before the tooltip was always shown of the right of the cursor and the cursor can't go off-screen, but now the tooltips are center-aligned with the button, meaning they can go off screen on the left if there is a button close to the left edge with a wide tooltip.

Reply:
Interesting approach. I'm okay with that, though. OK sign from me.